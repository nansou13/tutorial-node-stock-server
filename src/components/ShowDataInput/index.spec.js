import React from 'react'
import { shallow } from 'enzyme'
import { expect } from 'chai'
import renderer from 'react-test-renderer'
import ShowInput from '.'

const arrayValues = [
  {
    timestamp: 1576096456540,
    index: 1,
    stocks: {
      NASDAQ: 9.971326761277012,
      CAC40: 145.293302630023,
    },
  },
  {
    timestamp: 1576096457561,
    index: 2,
    stocks: {
      NASDAQ: 9.985167754304092,
      CAC40: 90.40672503192006,
    },
  },
  {
    timestamp: 1576096458561,
    index: 3,
    stocks: {
      NASDAQ: 9.981823896775474,
      CAC40: 90.39017756930647,
    },
  },
]

describe('<ShowInput />', () => {
  it('should render appropriately', () => {
    const handleChange = jest.fn()
    const handleFocus = jest.fn()
    const handleBlur = jest.fn()
    const props = {
      data: arrayValues,
      type: 'CAC40',
      handleChange,
      handleFocus,
      handleBlur,
      maxValue: 20,
    }
    const tree = renderer.create(<ShowInput {...props} />).toJSON()
    expect(tree).matchSnapshot()
  })
  it('should have x elements', () => {
    const handleChange = jest.fn()
    const handleFocus = jest.fn()
    const handleBlur = jest.fn()
    const props = {
      data: arrayValues,
      type: 'CAC40',
      handleChange,
      handleFocus,
      handleBlur,
      maxValue: 20,
    }
    const wrapper = shallow(<ShowInput {...props} />)
    expect(wrapper.find('td')).to.have.length(4)
  })
})
