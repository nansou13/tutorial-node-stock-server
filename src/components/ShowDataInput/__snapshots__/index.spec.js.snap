// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowInput /> should render appropriately 1`] = `
Array [
  <td>
    CAC40
  </td>,
  <td>
    <input
      onBlur={[MockFunction]}
      onChange={[Function]}
      onFocus={[MockFunction]}
      style={
        Object {
          "width": 55,
        }
      }
      type="number"
      value="145.293302630023"
    />
  </td>,
  <td>
    <input
      onBlur={[MockFunction]}
      onChange={[Function]}
      onFocus={[MockFunction]}
      style={
        Object {
          "width": 55,
        }
      }
      type="number"
      value="90.40672503192006"
    />
  </td>,
  <td>
    <input
      onBlur={[MockFunction]}
      onChange={[Function]}
      onFocus={[MockFunction]}
      style={
        Object {
          "width": 55,
        }
      }
      type="number"
      value="90.39017756930647"
    />
  </td>,
]
`;
